UM/Settings/SettingFunction.py

```diff
@@ -194,39 +194,153 @@ def visit_Name(self, node: ast.Name) -> None:  # [CodeStyle: ast.NodeVisitor req
         """
+        # The blacklisting is done just in case (All function calls should be whitelisted. The blacklist is to make
+        # extra sure that certain calls are *not* made!)
+        if node.s in self._blacklist:
+            raise IllegalMethodError(node.s)
+        if node.s.startswith("_"):
+            raise IllegalMethodError(node.s)
 
         if node.s not in self._knownNames and node.s not in dir(builtins):  # type: ignore #AST uses getattr stuff, so ignore type of node.s.
             self.keys.add(node.s)  # type: ignore
 
-    def visit_Constant(self, node: ast.AST) -> None:
+    def visit_Constant(self, node) -> None:
         """This one is used on Python 3.8+ to visit string types."""
-
+        # The blacklisting is done just in case (All function calls should be whitelisted. The blacklist is to make
+        # extra sure that certain calls are *not* made!)
+        if node.value in self._blacklist:
+            raise IllegalMethodError(node.value)
+        if node.s.startswith("_"):
+            raise IllegalMethodError(node.s)
         if isinstance(node.value, str) and node.value not in self._knownNames and node.value not in dir(builtins):  # type: ignore #AST uses getattr stuff, so ignore type of node.value.
             self.keys.add(node.value)  # type: ignore
 
     _knownNames = {
         "math",
+        "round",
         "max",
+        "ceil",
         "min",
+        "sqrt",
+        "log",
+        "tan",
+        "cos",
+        "sin",
+        "pi",
+        "floor"
         "debug",
         "sum",
         "len",
         "uuid",
         "hashlib",
-        "base64"
+        "base64",
+        "uuid3",
+        "NAMESPACE_DNS",
+        "decode",
+        "encode",
+        "b64encode",
+        "digest",
+        "md5",
+        "radians"
     }  # type: Set[str]
 
     _blacklist = {
         "sys",
         "os",
+        "delattr",
+        "getattr",
+        "dir",
+        "open",
+        "write",
+        "compile",
         "import",
         "__import__",
+        "__self__",
+        "_",  # Because you can use this guy to create a number of the other blacklisted strings
+        "__",
+        ".",
+        "_._",   # Just in case (I also don't see a reason for someone to use a string named like that...)
+        "__enter__",
+        "__builtins__",
         "eval",
         "exec",
+        "execfile",
+        "file",
         "subprocess",
+        "globals",
+        "__class__"
+        "__globals__"
+        "hasattr",
+        "raw_input",
+        "input",
+        "reload",
+        "setattr",
+        "vars",
+        "locals",
+        "system",
+        "literal_eval",
+        "ast.literal_eval",
+        "ast",
+        "lambda",
+        "__getattribute__",
+        "__setattr__",
+        "find_module",
+        "__pycache__",
+        "__file__"
     }  # type: Set[str]
+
+    _allowed_builtins = {
+        "bool",
```

Comment:
How about tuple?
How about sorted?
Hwo about range?
I could imagine use cases for those in setting functions.